### PR TITLE
Cpinitiative master

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

**Comment by nwatx** [comment]
wait woops i accidently created a pr

**Comment by maggie-j-liu** [comment]
image
where are the groups supposed to show? (i have admin perms and this is on the deployment)

**Comment by nwatx** [comment]
> image
> where are the groups supposed to show? (i have admin perms and this is on the deployment)

strange, on my emulator build it showed. Can you try signing out and back in?

**Comment by maggie-j-liu** [comment]
still doesn't show, but i can see admin settings so the permissions are working correctly
image

**Comment by nwatx** [comment]
wtmoo. can you try on this preview build once it finishes and ss console

**Comment by thecodingwizard** [comment]
what's up with package-lock and yarn lock? In general we should use yarn for everything for guide...

File: gatsby-config.js [hunk 1]
```diff
@@ -8,4 +8,8 @@ require('ts-node').register({
   },
 });
 
+require('dotenv').config({
+  path: `.env.${process.env.NODE_ENV}`,
+});
+
```
Comment (thecodingwizard) [inline gatsby-config.js:14]: oops wait why is this needed?

File: src/hooks/groups/useUserGroups.tsx [hunk 2]
```diff
@@ -46,25 +50,48 @@ const UserGroupsProvider = ({
...
         });
+
+        // if(permissions.isAdmin) return false;
       });
     },
-    [firebaseUser?.uid, updateCtr]
+    [firebaseUser?.uid, permissions, updateCtr]
   );
 
+  console.log(groups);
```
Comment (thecodingwizard) [inline src/hooks/groups/useUserGroups.tsx:93]: probs don't need console.logs

File: src/hooks/groups/useUserGroups.tsx [hunk 3]
```diff
@@ -46,25 +50,48 @@ const UserGroupsProvider = ({
...
+            ...doc.data(),
+          }));
+
+          // if all queries are done
           if (Object.keys(queries).every(x => queries[x] !== null)) {
-            setGroups(Object.values(queries).flat());
+            const queryResults = Object.values(queries).flat();
+            if (permissions.isAdmin)
+              // slice / 3 since there are 3 keys in object.keys that are redundant
+              setGroups(queryResults.slice(0, queryResults.length / 3));
```
Comment (thecodingwizard) [inline src/hooks/groups/useUserGroups.tsx:81]: don't query three times if admin. you should just query once only.

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
I updated firebase rules such that admins should have read access to everything. Haven't tested fully yet though.

In addition to review comments, I don't think any of the yarn.lock / package-lock.json files need to be touched.

Thanks for working on this! :)

**Comment by nwatx** [comment]
too many git errors, continued in new pr

**Pull request closed without merging**